obspy/taup/taup_geo.py

```diff
@@ -0,0 +1,159 @@
+        distance_in_km = values[0]/1000.0
+        # NB - km2deg assumes spherical Earth... generate a warning
+        if flattening_of_earth != 0.0:
+            msg = "Assuming spherical Earth when calculating angle. " + \
+                  "Install the Python module 'geographiclib' to solve this."
+            warnings.warn(msg)
+        distance_in_deg = kilometer2degrees(distance_in_km,
+                                            radius=radius_of_earth_in_km)
+    return distance_in_deg
+
+
+def add_geo_to_arrivals(arrivals, source_latitude_in_deg,
+                        source_longitude_in_deg, station_latitude_in_deg,
+                        station_longitude_in_deg, radius_of_earth_in_km,
+                        flattening_of_earth):
+    """
+    Add geographical information to arrivals
+
+    :param arrivals: Set of taup arrivals
+    :type: :class:`Arrivals`
+    :param source_latitude_in_deg: Source location latitude in degrees
+    :type source_latitude_in_deg: float
+    :param source_longitude_in_deg: Source location longitue in degrees
+    :type source_longitude_in_deg: float
+    :param station_latitude_in_deg: Station location latitude in degrees
+    :type station_latitude_in_deg: float
+    :param station_longitude_in_deg: Station location longitude in degrees
+    :type station_longitude_in_deg: float
+    :param radius_of_earth_in_km: Radius of the Earth in km
+    :type radius_of_earth_in_km: float
+    :param flattening_of_earth: Flattening of earth (0 for a sphere)
+    :type station_longitude_in_deg: float
+
+    :return: List of ``Arrival`` objects, each of which has the time,
+        corresponding phase name, ray parameter, takeoff angle, etc. as
+        attributes.
+    :rtype: :class:`Arrivals`
+    """
+    if HAS_GEOGRAPHICLIB:
+        ellipsoid = Geodesic(a=radius_of_earth_in_km * 1000.0,
+                             f=flattening_of_earth)
+        g = ellipsoid.Inverse(source_latitude_in_deg, source_longitude_in_deg,
+                              station_latitude_in_deg,
+                              station_longitude_in_deg)
+        azimuth = g['azi1']
+        line = ellipsoid.Line(source_latitude_in_deg,
+                              source_longitude_in_deg, azimuth)
+
+        # We may need to update many arrival objects
+        # and each could have pierce points and a
+        # path
+        for arrival in arrivals:
+
+            if arrival.pierce is not None:
+                pathList = []
+                for pierce_point in arrival.pierce:
+                    pos = line.ArcPosition(np.degrees(pierce_point['dist']))
+                    diffTDG = np.array([(
+                        pierce_point['p'],
+                        pierce_point['time'],
+                        pierce_point['dist'],
+                        pierce_point['depth'],
+                        pos['lat2'],
+                        pos['lon2'])], dtype=TimeDistGeo)
+                    pathList.append(diffTDG)
```

Comment:
You already have the length of `arrival.pierce`; you should pre-allocate the replacement before the loop instead of re-allocating each time.